metaflow/plugins/aws/batch/batch_client.py

```diff
@@ -575,6 +582,28 @@ def __init__(self, group, stream, pos=0, sleep_on_no_data=0):
         self._token = None
 
     def _get_events(self):
+        done = RunningJob.is_done
```

Comment:
is_done and status_code are instance level properties so you can't call them on a class object.